crates/autopilot/src/domain/settlement/observer.rs

```diff
@@ -172,4 +234,88 @@ impl Observer {
+                return Err(anyhow!(format!(
+                    "node could not find the transaction - tx: {tx:?}, err: {err:?}",
+                )));
+            }
+        };
+
+        match transaction {
+            Ok(multi_transaction) => {
+                let mut settlements = Vec::new();
+                
+                for settlement_data in &multi_transaction.settlements {
+                    let single_transaction = settlement::Transaction {
+                        hash: multi_transaction.hash,
+                        auction_id: settlement_data.auction_id,
+                        block: multi_transaction.block,
+                        timestamp: multi_transaction.timestamp,
+                        gas: multi_transaction.gas,
+                        gas_price: multi_transaction.gas_price,
+                        solver: multi_transaction.solver,
+                        trades: settlement_data.trades.clone(),
+                    };
+
+                    match settlement::Settlement::new(single_transaction, &self.persistence, self.eth.chain())
+                        .await
+                    {
+                        Ok(settlement) => {
+                            settlements.push(settlement);
+                        }
+                        Err(settlement::Error::Infra(err)) => {
+                            // bubble up retryable error
+                            return Err(err);
+                        }
+                        Err(err) => {
+                            tracing::warn!(?tx, ?settlement_data.auction_id, ?err, "invalid settlement in multi-settlement transaction");
+                            // Continue processing other settlements even if one fails
+                        }
+                    }
+                }
+
+                Ok(settlements)
+            }
+            Err(err) => {
+                match err {
+                    settlement::transaction::Error::MissingCalldata => {
+                        tracing::error!(?tx, ?err, "invalid settlement transaction");
+                        Ok(vec![])
+                    }
+                    settlement::transaction::Error::MissingAuctionId
+                    | settlement::transaction::Error::Decoding(_)
+                    | settlement::transaction::Error::SignatureRecover(_)
+                    | settlement::transaction::Error::OrderUidRecover(_)
+                    | settlement::transaction::Error::MissingSolver => {
+                        tracing::warn!(?tx, ?err, "invalid settlement transaction");
+                        Ok(vec![])
+                    }
+                    settlement::transaction::Error::Authentication(_) => {
+                        // This has to be a temporary error because the settlement contract
+                        Err(anyhow!(format!(
+                            "could not determing solver address - err: {err:?}"
+                        )))
+                    }
```

Comment:
```suggestion
                    settlement::transaction::Error::Authentication(_) => {
                        // This has to be a temporary error because the settlement contract
                        tracing::warn!(?tx, ?err, "could not determine solver address");
                        Err(anyhow!(format!(
                            "could not determine solver address - err: {err:?}"
                        )))
                    }
```